classes/ViewFactory.php

```diff
@@ -33,10 +33,25 @@ public static function create(ContainerInterface $container, Request $request =
         }
 
         $view = new Smarty(__DIR__ . '/../templates/');
+
+        $uri = $request->getUri();
         if (in_array('https', $request->getHeader('X-Forwarded-Proto'))) {
-            $request = $request->withUri($request->getUri()->withScheme('https')->withPort(443));
+            $uri = $uri->withScheme('https')->withPort(443);
+        }
+
+        $port = current($request->getHeader('X-Forwarded-Port'));
+        if ($port) {
+            $uri = $uri->withPort(intVal($port));
         }
 
+        $path = current($request->getHeader('X-Forwarded-Path'));
+        if ($path) {
+            $uri = $uri->withBasePath($path);
+        }
+
+        $request = $request->withUri($uri);
```

Comment:
I just noticed we don't need to keep the `$request` object and can use `$uri` directly:
```diff
diff --git a/classes/ViewFactory.php b/classes/ViewFactory.php
index a2a05b1..63f7fe9 100644
--- a/classes/ViewFactory.php
+++ b/classes/ViewFactory.php
@@ -49,13 +49,10 @@ class ViewFactory
             $uri = $uri->withBasePath($path);
         }
 
-        $request = $request->withUri($uri);
-
-
         /** @var LocaleManager $localeManager */
         $localeManager = $container->get('locale');
 
-        $smartyPlugins = new SmartyPlugins($container->get('router'), $request->getUri()->withUserInfo(null));
+        $smartyPlugins = new SmartyPlugins($container->get('router'), $uri->withUserInfo(null));
         $view->registerPlugin('function', 'path_for', [$smartyPlugins, 'pathFor']);
         $view->registerPlugin('function', 'base_url', [$smartyPlugins, 'baseUrl']);
         $view->registerPlugin('block', 't', [$localeManager, 'smartyTranslate']);
```

Reply:
done